src/typings/index.ts

```diff
@@ -1,39 +1,30 @@
 import { Permissions } from './permissions';
-import { Component, ComponentType, ComponentValue, FacetConfig, FacetConfigType } from './schema';
+import {
+    ComponentConfig, ComponentType, FacetConfig, FacetConfigType,
+    ValueReference
+} from './schema';
 import { ComponentClass, SFC } from 'react';
 
 export interface KeyValueObject {
     [name: string]: string;
 }
 
-export interface ValueItem {
-    value: ComponentValue;
-    name: string;
-}
-
 // base type of NormalizedComponent & NormalizedFacetConfig
 export interface NormalizedItem {
     type: FacetConfigType | ComponentType;
     id: number;
 }
 
-export type NormalizedComponent = NormalizedItem & Component & {
+export type NormalizedComponent = NormalizedItem & ComponentConfig & {
```

Comment:
Moet `NormalizedComponent` niet naar `NormalizedComponentConfig` hernoemt worden?
Beetje lange naam, I know, maar aangezien we ook al `NormalizedFacetConfig` hebben.